lib/libspl/include/sys/spa_os.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+/*
+ * Copyright (c) 2011, 2019 by Delphix. All rights reserved.
+ */
+
+#ifndef _SYS_SPA_OS_H
+#define	_SYS_SPA_OS_H
+
+typedef struct spa_history_list {
+	uint64_t		size;
+	procfs_list_t		procfs_list;
```

Comment:
Could you instead just borrow the user space `procfs` code from `lib/libzpool/kernel.c` and add it to the FreeBSD SPL?  It does very little and would only cost you a little memory to maintain the structure.  Adding that code seems like it would let us avoid a lot of additional refactoring.

Reply:
ok - I'll try